### PR TITLE
Correct channels_first format in 1d_pooling for tf.layers

### PR DESCRIPTION
**Pull request by vroger11**
The code was equivalent in the case of self.data_format == 'channels_last' or self.data_format == 'channels_first'.
My modifications fix it when data_format == 'channels_first'.

### CODE REVIEW

**Comment by vroger11** [comment]
I signed it!

**Comment by martinwicke** [comment]
This change is good, I think. But the fact that this survived indicates there's not enough test coverage of this code. Could you add a test which exercises the `channels_first` code path to make sure it works?

**Comment by vroger11** [comment]
I am not familiar with testing procedures in tensorflow and do not have much time to do it.
Whereas this bug is problematic and should be fixed because it will not generate an exception and will  use the wrong data_format.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by rmlarsen** [comment]
@vroger11 @martinwicke how to proceed. I don't think we should merge this without test coverage.

**Comment by martinwicke** [comment]
Fine. I don't like no tests, but I like it better than having no tests and being broken.

**Comment by martinwicke** [comment]
Argh, this PR is against 1.4. That's bad. Can you make a new PR against master? We don't take PRs against release branches.
